tests/Carbon/CreateSafeTest.php

```diff
@@ -26,10 +26,7 @@ public function testInvalidDateExceptionProperties()
 
     public function testCreateSafeThrowsExceptionForSecondLowerThanZero()
     {
-        $this->expectException(InvalidDateException::class);
-        $this->expectExceptionMessage(
-            'second : -1 is not a valid value.'
-        );
+        $this->expectExceptionObject(new InvalidDateException('second', -1));
```

Comment:
Sounds to me like we testing here `2 + 2 = 2 + 2` instead of testing `2 + 2 = 4` when doing this we mirror a little more the source code and we'll no longer detect if the `InvalidDateException` code/message broke/changed. I have to double-think of that.

Reply:
You will because, because there is a custom `\Tests\Carbon\Exceptions\InvalidDateExceptionTest::testInvalidCastException` will fail.

I like to think this is a command bus testing approach.
- we test that the command is called with the expected argument (how is the exception constructed?)
- we ensure that the handler does the expected thing (what will be the built message of the exception depending on its arguments)

Reply:
Nice, indeed it's a good approach.